src/StepAggregator.php

```diff
@@ -192,4 +159,35 @@ public function getName()
     {
         return $this->name;
     }
+
+    /**
+     * Builds the pipeline
+     *
+     * @return callable
+     */
+    private function buildPipeline()
```

Comment:
This build pipeline method is only there because writers and other steps (like array check (which should go away if items are introduced) or pcntl) must be added to the pipeline. The reason why they should NOT be extracted to a builder is that the pipeline must be built again between calling `process`. The following scenario is very unlikely, but possible:

``` php
$workflow->addStep(...);
//other steps added
$workflow->process();
//further steps added
$workflow->process();
```